### PR TITLE
Remove unnecessary lcov-result-merger dep from @firebase/messaging

### LINKED ISSUE #546: firebase doesn't play nicely with `yarn install --flat`
### [REQUIRED] Describe your environment

  * Operating System version: Linux 4.15.7 (Solus 3)
  * Firebase SDK version: npm firebase@4.10.1
  * Firebase Product: firestore

### [REQUIRED] Describe the problem

I was trying to follow best practices with my web app and ran `yarn install --prod --flat` to make sure that I wasn't using delivering multiple copies of the same libraries to my clients.  I found several libraries with conflicts and they were all being pulled in by firebase.  Specifically: minimist, readable-stream, isarray, process-nextick-args, string_decoder, assert-plus, extsprintf, glob, through2, vinyl, clone, clone-stats, replace-ext, glob-parent, is-glob, is-extglob, kind-of, and is-number. 

Some further debugging...
```bash
$ npm ls --prod minimist readable-stream isarray process-nextick-args string_decoder assert-plus extsprintf glob through2 vinyl clone clone-stats replace-ext glob-parent is-glob is-extglob kind-of is-number
site@1.0.0 /home/user/workspace/site
└─┬ firebase@4.10.1
  ├─┬ @firebase/firestore@0.3.4
  │ └─┬ grpc@1.9.1
  │   ├─┬ node-pre-gyp@0.6.39
  │   │ ├─┬ mkdirp@0.5.1
  │   │ │ └── minimist@0.0.8
  │   │ ├─┬ npmlog@4.1.2
  │   │ │ └─┬ are-we-there-yet@1.1.4
  │   │ │   └── readable-stream@2.3.3  deduped
  │   │ ├─┬ rc@1.2.4
  │   │ │ └── minimist@1.2.0
  │   │ ├─┬ request@2.81.0
  │   │ │ └─┬ http-signature@1.1.1
  │   │ │   ├── assert-plus@0.2.0
  │   │ │   ├─┬ jsprim@1.4.1
  │   │ │   │ ├── assert-plus@1.0.0
  │   │ │   │ ├── extsprintf@1.3.0
  │   │ │   │ └─┬ verror@1.10.0
  │   │ │   │   ├── assert-plus@1.0.0
  │   │ │   │   └── extsprintf@1.3.0  deduped
  │   │ │   └─┬ sshpk@1.13.1
  │   │ │     ├── assert-plus@1.0.0
  │   │ │     ├─┬ dashdash@1.14.1
  │   │ │     │ └── assert-plus@1.0.0
  │   │ │     └─┬ getpass@0.1.7
  │   │ │       └── assert-plus@1.0.0
  │   │ ├─┬ rimraf@2.6.2
  │   │ │ └── glob@7.1.2
  │   │ └─┬ tar-pack@3.4.1
  │   │   └─┬ readable-stream@2.3.3
  │   │     ├── isarray@1.0.0
  │   │     ├── process-nextick-args@1.0.7
  │   │     └── string_decoder@1.0.3
  │   └─┬ protobufjs@5.0.2
  │     └── glob@7.1.2
  └─┬ @firebase/messaging@0.2.1
    └─┬ lcov-result-merger@2.0.0
      ├─┬ through2@2.0.3
      │ └─┬ readable-stream@2.3.5
      │   ├── isarray@1.0.0
      │   ├── process-nextick-args@2.0.0
      │   └── string_decoder@1.0.3
      ├─┬ vinyl@2.1.0
      │ ├── clone@2.1.1
      │ ├── clone-stats@1.0.0
      │ ├─┬ cloneable-readable@1.0.0
      │ │ ├── process-nextick-args@1.0.7
      │ │ └── through2@2.0.3  deduped
      │ └── replace-ext@1.0.0
      └─┬ vinyl-fs@2.4.4
        ├─┬ duplexify@3.5.4
        │ └── readable-stream@2.3.5  deduped
        ├─┬ glob-stream@5.3.5
        │ ├── glob@5.0.15
        │ ├─┬ glob-parent@3.1.0
        │ │ └─┬ is-glob@3.1.0
        │ │   └── is-extglob@2.1.1
        │ ├─┬ micromatch@2.3.11
        │ │ ├─┬ braces@1.8.5
        │ │ │ └─┬ expand-range@1.8.2
        │ │ │   └─┬ fill-range@2.2.3
        │ │ │     ├─┬ is-number@2.1.0
        │ │ │     │ └── kind-of@3.2.2  deduped
        │ │ │     ├─┬ isobject@2.1.0
        │ │ │     │ └── isarray@1.0.0  deduped
        │ │ │     └─┬ randomatic@1.1.7
        │ │ │       ├─┬ is-number@3.0.0
        │ │ │       │ └── kind-of@3.2.2
        │ │ │       └── kind-of@4.0.0
        │ │ ├─┬ extglob@0.3.2
        │ │ │ └── is-extglob@1.0.0
        │ │ ├── is-extglob@1.0.0
        │ │ ├─┬ is-glob@2.0.1
        │ │ │ └── is-extglob@1.0.0  deduped
        │ │ ├── kind-of@3.2.2
        │ │ └─┬ parse-glob@3.0.4
        │ │   ├─┬ glob-base@0.3.0
        │ │   │ ├─┬ glob-parent@2.0.0
        │ │   │ │ └── is-glob@2.0.1  deduped
        │ │   │ └─┬ is-glob@2.0.1
        │ │   │   └── is-extglob@1.0.0
        │ │   ├── is-extglob@1.0.0
        │ │   └─┬ is-glob@2.0.1
        │ │     └── is-extglob@1.0.0  deduped
        │ ├─┬ ordered-read-streams@0.3.0
        │ │ └── readable-stream@2.3.5  deduped
        │ └─┬ through2@0.6.5
        │   └─┬ readable-stream@1.0.34
        │     ├── isarray@0.0.1
        │     └── string_decoder@0.10.31
        ├─┬ gulp-sourcemaps@1.6.0
        │ ├── through2@2.0.3  deduped
        │ └─┬ vinyl@1.2.0
        │   ├── clone@1.0.3
        │   ├── clone-stats@0.0.1
        │   └── replace-ext@0.0.1
        ├─┬ lazystream@1.0.0
        │ └── readable-stream@2.3.5  deduped
        ├─┬ merge-stream@1.0.1
        │ └── readable-stream@2.3.5  deduped
        ├─┬ mkdirp@0.5.1
        │ └── minimist@0.0.8
        ├── readable-stream@2.3.5  deduped
        ├── through2@2.0.3  deduped
        ├─┬ through2-filter@2.0.0
        │ └── through2@2.0.3  deduped
        └─┬ vinyl@1.2.0
          ├── clone@1.0.3
          ├── clone-stats@0.0.1
          └── replace-ext@0.0.1

```

As you can see, all of these conflicts are getting pulled in due to `grpc` getting pulled in by `@firebase/firestore` and `lcov-result-merger` getting pulled in by `@firebase/messaging`.  

The `grpc` package ultimately gets tree-shaken out on non-node platforms, so it really isn't a problem.  However, since it's listed as a normal dependency, yarn can't know that it will ultimately get excluded from the build and so it gets included in the --flat resolution.  I'm not sure what the best solution to this problem is, but I suspect the easiest thing would be to make it an optional dependency.

The `lcov-result-merger` package appears to have been added as part of [this commit](https://github.com/firebase/firebase-js-sdk/commit/dd3751ec0951d042413416b2d2fd812d12aae285#diff-d9729f3bea478cb934dc8f484a5c9f2b) by mistake.  It's only used as a cli tool during development and it's already included as a dev dependency at the root of the project.  I removed it from `@firebase/messaging` and all the tests still pass.

#### Steps to reproduce:
```
#vue isn't necessary to reproduce this.  This was just one of the simplest demos I could find that actually imported firebase as an npm package 
git clone git@github.com:Q42/vuejs-firestore-demo.git 
cd vuejs-firestore-demo
npm upgrade firebase

# then, to see the issue before we get yarn involved
npm ls --prod minimist readable-stream isarray process-nextick-args string_decoder assert-plus extsprintf glob through2 vinyl clone clone-stats replace-ext glob-parent is-glob is-extglob kind-of is-number

# to recreate the issue itself, run
yarn install --prod --flat
```

#### Relevant Code:

see reproduction steps

### PR DESCRIPTION
**Pull request by zevdg**
tentatively fixes #546 

I'm not sure that making grpc an optional dependency is the best way to fix this, but I figured I'd create this PR as a starting point.

### CODE REVIEW

File: packages/messaging/package.json [hunk 1]
```diff
@@ -18,7 +18,6 @@
   "dependencies": {
     "@firebase/messaging-types": "0.1.2",
     "@firebase/util": "0.1.10",
-    "lcov-result-merger": "^2.0.0",
```
Comment (gauntface) [inline packages/messaging/package.json]: @jshcrowthe not sure why this is in dependencies but might be needed in devdeps?

Comment (jshcrowthe) [reply packages/messaging/package.json]: This actually isn't needed at all in this package. We use it at a higher level. I seems it snuck in, in #482

**Comment by jshcrowthe** [comment]
Per comments in #546, I don't think the changes related to `grpc` are something we want to keep. That said, the change related to the `lcov-result-merger` package is something we'll need to do and I'm happy to get you contributor cred 😄 !

Can we refactor this PR to remove only that dep?

**Comment by zevdg** [comment]
there you go.

**Pull request merged**